core/types/calls.cc

```diff
@@ -2989,7 +2991,8 @@ class Magic_splat : public IntrinsicMethod {
         if (!dispatched.main.errors.empty()) {
             // In case of an error, the splat is converted to an array with a single
             // element; be conservative in what we declare the element type to be.
-            res.returnType = Types::arrayOfUntyped();
+            // TODO(neil): this should actually blame to an error recovery magic symbol
```

Comment:
When it says "in case of an error," what it's referring to is the fake dispatch to `.to_a` that happens a few lines above. Sorbet never reports those errors to the user, so this is not an error recovery case.

Rather, for this we do want the untyped to blame to the Magic_splat symbol, like you've done it. Knowing that untyped are coming from this would let us know whether we wanted to look at better typing support for splats.